tools/mtouch/mtouch.cs

```diff
@@ -1323,10 +1323,22 @@ static int Main2 (string[] args)
 			if (app.EnableRepl && app.LinkMode != LinkMode.None)
 				throw new MonoTouchException (82, true, "REPL (--enable-repl) is only supported when linking is not used (--nolink).");
 
-			// needs to be set after the argument validations
-			// interpreter can use some extra code (e.g. SRE) that is not shipped in the default (AOT) profile
-			if (app.UseInterpreter)
+			if (app.UseInterpreter) {
+				// it's confusing to use different options to get a feature to work (e.g. dynamic, SRE...) on both simulator and device
+				if (app.IsSimulatorBuild) {
+					ErrorHelper.Show (ErrorHelper.CreateWarning (141, "The interpreter is not supported in the simulator. Switching to REPL which provide the same extra features on the simulator."));
+					app.UseInterpreter = false;
+				}
+
+				// FIXME: the interpreter only supports ARM64 right now
+				// temporary - without a check here the error happens when deploying
+				if (!app.IsArchEnabled (Abi.ARM64))
+					throw ErrorHelper.CreateError (99, "Internal error: The interpreter is currently only available for 64 bits.");
```

Comment:
Doesn't this error trigger when passing `--interpreter` when building for the simulator?

Reply:
yup, made that one before enabling sim :)
there's no point in limiting the sim to 64 bits (since it's not interpreted)
Fix incoming...